### PR TITLE
feat: [SIW-1817] Add Wallet Instance attested key validation event based

### PR DESCRIPTION
**Pull request by grausof**
It introduces the verification of the validity of the certified key inside the **Wallet Instances** released for Android and proceeds to the revocation in case the certificates are no longer valid.

In particular, two functions are added:
- `addWalletInstanceToValidationQueue` triggered by cosmos that filters the new wallet instances based on the device and the presence of the certificate and adds them to a queue

- `validateWalletInstance` which is triggered by the above queue and proceeds to the validation and if necessary the revocation of the Wallet Instance. If the Wallet Instance must not be revoked it is reinserted in the queue with a `visibilityTimeout` of 24 hours.

### CODE REVIEW

**Comment by grausof** [comment]
@Krusty93 Please check the terraform because I'm not sure it's done well

File: apps/io-wallet-user-func/src/app/main.ts [hunk 1]
```diff
@@ -162,3 +180,33 @@ app.http("setCurrentWalletInstanceStatus", {
...
   route: "wallet-instances/current/status",
 });
+
+app.cosmosDB("addWalletInstanceToValidationQueue", {
+  connection: "CosmosDbEndpoint",
+  containerName: "wallet-instances",
+  createLeaseContainerIfNotExists: true,
+  databaseName: config.azure.cosmos.dbName,
+  handler: AddWalletInstanceToValidationQueueFunction({
+    inputDecoder: t.array(WalletInstance),
```
[the remark this answers is not in the document]
Comment (grausof) [reply apps/io-wallet-user-func/src/app/main.ts:190]: Added thanks

File: infra/resources/_modules/storage/storage_accounts.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+data "azurerm_storage_account" "wallet_revocation_storage_data" {
+  name                = azurerm_storage_account.wallet_revocation_storage.name
+  resource_group_name = var.resource_group_name
+}
+
+resource "azurerm_key_vault_secret" "wallet_revocation_storage_connection_string" {
+  name         = "WalletRevocationStorageConnectionString"
+  value        = data.azurerm_storage_account.wallet_revocation_storage_data.primary_connection_string
+  key_vault_id = var.key_vault_wallet_id
+}
```
Comment (silvicir) [inline infra/resources/_modules/storage/storage_accounts.tf:23]:
```suggestion
resource "azurerm_key_vault_secret" "wallet_revocation_storage_connection_string" {
  name  = "WalletRevocationStorageConnectionString"
  value = azurerm_storage_account.wallet_revocation_storage.primary_connection_string
  key_vault_id = var.key_vault_wallet_id
}
```

File: apps/io-wallet-user-func/src/infra/azure/functions/add-wallet-instance-to-validation-queue.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { AddWalletInstanceToValidationQueueHandler } from "@/infra/handlers/add-wallet-instance-to-validation-queue";
+import { azureFunction } from "@pagopa/handler-kit-azure-func";
+import { pipe } from "fp-ts/function";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import { sendTelemetryException } from "io-wallet-common/infra/azure/appinsights/telemetry";
+
+export const AddWalletInstanceToValidationQueueFunction = azureFunction(
+  pipe(
+    AddWalletInstanceToValidationQueueHandler,
+    RTE.orElseFirstW((error) =>
```
Comment (silvicir) [inline apps/io-wallet-user-func/src/infra/azure/functions/add-wallet-instance-to-validation-queue.ts:10]: for consistency with the other functions, we should place the telemetry part in the handlers

File: apps/io-wallet-user-func/src/infra/azure/functions/validate-wallet-instance-attested-key.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { ValidateWalletInstanceAttestedKeyHandler } from "@/infra/handlers/validate-wallet-instance-attested-key";
+import { azureFunction } from "@pagopa/handler-kit-azure-func";
+import { pipe } from "fp-ts/function";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import { sendTelemetryException } from "io-wallet-common/infra/azure/appinsights/telemetry";
+
+export const ValidateWalletInstanceAttestedKeyFunction = azureFunction(
+  pipe(
+    ValidateWalletInstanceAttestedKeyHandler,
+    RTE.orElseFirstW((error) =>
```
Comment (silvicir) [inline apps/io-wallet-user-func/src/infra/azure/functions/validate-wallet-instance-attested-key.ts:10]: ditto

File: apps/io-wallet-user-func/src/wallet-instance-revocation-process.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+} from "./wallet-instance";
+
+const validateWalletInstanceCertificatesChain =
+  (attestationServiceConfiguration: AttestationServiceConfiguration) =>
+  (
+    walletInstance: WalletInstanceValidWithAndroidCertificatesChain,
+  ): TE.TaskEither<Error, ValidationResult> => {
+    const { androidCrlUrl, googlePublicKey, httpRequestTimeout } =
+      attestationServiceConfiguration;
+    return pipe(
```
Comment (silvicir) [inline apps/io-wallet-user-func/src/wallet-instance-revocation-process.ts:30]:
```suggestion
const validateWalletInstanceCertificatesChain =
  ({ androidCrlUrl, googlePublicKey, httpRequestTimeout }: AttestationServiceConfiguration) =>
  (
    walletInstance: WalletInstanceValidWithAndroidCertificatesChain,
  ): TE.TaskEither<Error, ValidationResult> =>
   pipe(
```

**Review by silvicir — APPROVED** [review]

File: infra/resources/_modules/storage/storage_accounts.tf [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  account_tier              = "Standard"
+  account_kind              = "StorageV2"
+  account_replication_type  = "ZRS"
+  shared_access_key_enabled = false
+  min_tls_version           = "TLS1_2"
+
+  tags = var.tags
+}
+
+resource "azurerm_key_vault_secret" "wallet_revocation_storage_connection_string" {
```
Comment (Krusty93) [inline infra/resources/_modules/storage/storage_accounts.tf:14]: Please run pre-commit here

**Review by Krusty93 — APPROVED** [review]

**Pull request merged**